File: leetcode-solutions/0121-best-time-to-buy-and-sell-stock/solution.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int min_price = INT_MAX;
        int max_profit = 0;
        for(int x:prices){
            if(x>min_price){
                max_profit = max(max_profit,x-min_price);
            }
            min_price = min(x,min_price);
        }
        return max_profit;
    }
};

File: leetcode-solutions/0121-best-time-to-buy-and-sell-stock/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min_price = float('inf')
        max_profit = 0
        for n in prices:
            if n<min_price:
                min_price = n
            profit = n-min_price
            if profit>max_profit:
                max_profit = profit 
        return max_profit 

